hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java

```diff
@@ -0,0 +1,456 @@
+import com.hazelcast.scheduledexecutor.ScheduledTaskHandler;
+import com.hazelcast.scheduledexecutor.ScheduledTaskStatistics;
+import com.hazelcast.scheduledexecutor.StaleTaskException;
+import com.hazelcast.scheduledexecutor.StatefulTask;
+import com.hazelcast.scheduledexecutor.impl.operations.ResultReadyNotifyOperation;
+import com.hazelcast.scheduledexecutor.impl.operations.SyncStateOperation;
+import com.hazelcast.spi.InvocationBuilder;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.OperationService;
+import com.hazelcast.spi.impl.executionservice.InternalExecutionService;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.hazelcast.scheduledexecutor.impl.DistributedScheduledExecutorService.MEMBER_PARTITION;
+import static com.hazelcast.scheduledexecutor.impl.DistributedScheduledExecutorService.SERVICE_NAME;
+import static com.hazelcast.util.ExceptionUtil.sneakyThrow;
+
+public class ScheduledExecutorContainer {
+
+    private final ILogger logger;
+
+    private final String name;
+
+    private final NodeEngine nodeEngine;
+
+    private final InternalExecutionService executionService;
+
+    private final int partitionId;
+
+    private final int durability;
+
+    private final ConcurrentMap<String, ScheduledTaskDescriptor> tasks;
+
+    private final Map<String, BackupTaskDescriptor> backups;
+
+    private final AtomicBoolean memberPartitionLock = new AtomicBoolean();
+
+    public ScheduledExecutorContainer(String name, int partitionId, NodeEngine nodeEngine,
+                                      int durability) {
+        this(name, partitionId, nodeEngine, durability, new ConcurrentHashMap<String, BackupTaskDescriptor>());
+    }
+
+    public ScheduledExecutorContainer(String name, int partitionId, NodeEngine nodeEngine,
+                                      int durability, Map<String, BackupTaskDescriptor> backups) {
+        this.logger = nodeEngine.getLogger(getClass());
+        this.name = name;
+        this.nodeEngine = nodeEngine;
+        this.executionService = (InternalExecutionService) nodeEngine.getExecutionService();
+        this.partitionId = partitionId;
+        this.durability = durability;
+        this.tasks = new ConcurrentHashMap<String, ScheduledTaskDescriptor>();
+        this.backups = backups;
+    }
+
+    public <V> ScheduledFuture<V> schedule(TaskDefinition definition) {
+        try {
+            acquireMemberPartitionLockIfNeeded();
```

Comment:
I'm ok with this solution but I think creating a separate class for member-container would be better. We would decouple partition and member logic. 
getting rid off these `if else` blocks would be the extra, we had them all over the class
```
if member-partition 
...
else
...
```